src/autocomplete/inline_screenreader.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+
+/**
+ * This object is used to communicate inline code completions rendered into an editor with ghost text to screen reader users.
+ */
+class AceInlineScreenReader {
+    /**
+     * Creates the off-screen div in which the ghost text content in redered and which the screen reader reads.
+     * @param {import("../editor").Editor} editor
+     */
+    constructor(editor) {
+        this.editor = editor;
+
+        this.screenReaderDiv = document.createElement("div");
+        this.screenReaderDiv.classList.add("ace_screenreader-only");
+        this.editor.container.appendChild(this.screenReaderDiv);
+    }
+
+    /**
+     * Set the ghost text content to the screen reader div
+     * @param {string} content
+     */
+    setScreenReaderContent(content) {
+        // Path for when inline preview is used with 'normal' completion popup.
+        if (!this.popup && this.editor.completer && this.editor.completer.popup) {
+            this.popup = this.editor.completer.popup;
+
+            this.popup.renderer.on("afterRender", function() {
+                let row = this.popup.getRow();
+                let t = this.popup.renderer.$textLayer;
+                let selected = t.element.childNodes[row - t.config.firstRow];
+                if (selected) {
+                    let idString = "doc-tooltip ";
+                    for (let lineIndex = 0; lineIndex < this._lines.length; lineIndex++) {
+                        idString += `ace-inline-screenreader-line-${lineIndex} `;
+                    }
+                    selected.setAttribute("aria-describedby", idString);      
+                }
+            }.bind(this));
+        }
+
+        // TODO: Path for when special inline completion popup is used.
+
+        // Remove all children of the div
+        while (this.screenReaderDiv.firstChild) {
+            this.screenReaderDiv.removeChild(this.screenReaderDiv.firstChild);
+        }
+        this._lines = content.split(/\r\n|\r|\n/);
+        const codeElement = this.createCodeBlock();
+        this.screenReaderDiv.appendChild(codeElement);
+    }
+
+    destroy() {
+        // Remove all children of the div
+        while (this.screenReaderDiv.firstChild) {
```

Comment:
Is this needed when you remove the div itself below?

Reply:
What happens if you remove an element which has child nodes? (I was expecting bad things to happen, which is why I removed the child nodes first here)

Reply:
They will be removed as well.